crates/rustc_codegen_spirv/src/symbols.rs

```diff
@@ -254,61 +407,70 @@ pub fn parse_attrs(
+        attrs.iter().flat_map(|attr| {
+            let is_spirv = match attr.kind {
+                AttrKind::Normal(ref item) => {
+                    // TODO: We ignore the rest of the path. Is this right?
+                    let last = item.path.segments.last();
+                    last.map_or(false, |seg| seg.ident.name == cx.sym.spirv)
                 }
+                AttrKind::DocComment(..) => false,
+            };
+            let args = if !is_spirv {
+                // Use an empty vec here to return empty
+                Vec::new()
+            } else if let Some(args) = attr.meta_item_list() {
+                args
             } else {
-                let name = match arg.ident() {
-                    Some(i) => i,
-                    None => {
-                        cx.tcx.sess.span_err(
-                            arg.span(),
-                            "#[spirv(..)] attribute argument must be single identifier",
-                        );
-                        return None;
+                cx.tcx.sess.span_err(
+                    attr.span,
+                    "#[spirv(..)] attribute must have at least one argument",
+                );
+                Vec::new()
+            };
+            args.into_iter().filter_map(move |ref arg| {
+                if arg.has_name(cx.sym.really_unsafe_ignore_bitcasts) {
+                    Some(SpirvAttribute::ReallyUnsafeIgnoreBitcasts)
+                } else if arg.has_name(cx.sym.descriptor_set) {
+                    match parse_attr_int_value(cx, arg) {
+                        Some(x) => Some(SpirvAttribute::DescriptorSet(x)),
+                        None => None,
                     }
-                };
-                match cx.sym.attributes.get(&name.name) {
-                    Some(a) => Some(a.clone()),
-                    None => {
-                        cx.tcx
-                            .sess
-                            .span_err(name.span, "unknown argument to spirv attribute");
-                        None
+                } else if arg.has_name(cx.sym.binding) {
+                    match parse_attr_int_value(cx, arg) {
+                        Some(x) => Some(SpirvAttribute::Binding(x)),
+                        None => None,
                     }
+                } else {
+                    let name = match arg.ident() {
+                        Some(i) => i,
+                        None => {
+                            cx.tcx.sess.span_err(
+                                arg.span(),
+                                "#[spirv(..)] attribute argument must be single identifier",
+                            );
+                            return None;
+                        }
+                    };
+                    cx.sym
+                        .attributes
+                        .get(&name.name)
+                        .map(|a| match a {
+                            SpirvAttribute::Entry(entry) => SpirvAttribute::Entry(
```

Comment:
I would prefer this gets handled in the same way `descriptor_set` and `binding` is handled, but whatever, that can be cleaned up later.

Reply:
So you're saying add additional if else statements for each entry / execution_model? Wouldn't putting everything in the attributes HashMap scale better?